charon/src/remove_read_discriminant.rs

```diff
@@ -48,67 +48,74 @@ impl<'a, 'tcx, 'ctx> Visitor<'a, 'tcx, 'ctx> {
-                let covers_all = {
-                    // Lookup the type of the scrutinee
-                    match self.ctx.type_decls.get(*adt_id) {
-                        None => {
-                            // This can happen if there was an error while
-                            // extracting the definitions
-                            assert!(self.ctx.error_count > 0);
-                            // For safety, we consider that not all the patterns
-                            // were covered
-                            false
-                        }
-                        Some(d) => {
-                            match &d.kind {
-                                TypeDeclKind::Struct(_) | TypeDeclKind::Opaque => {
-                                    // We shouldn't get there
-                                    register_error_or_panic!(
-                                        self.ctx,
-                                        st.meta.span.rust_span_data.span(),
-                                        "Unreachable case"
-                                    );
-                                    false
-                                }
-                                TypeDeclKind::Error(_) => false,
-                                TypeDeclKind::Enum(variants) => {
-                                    // Check that all the variants are covered
-                                    variants
-                                        .iter()
-                                        .enumerate()
-                                        .all(|(i, _)| covered_variants.contains(&i))
-                                }
+                // Lookup the type of the scrutinee
+                let variants = match self.ctx.type_decls.get(*adt_id) {
+                    // This can happen if there was an error while extracting the definitions
+                    None => None,
+                    Some(d) => {
+                        match &d.kind {
+                            TypeDeclKind::Struct(_) | TypeDeclKind::Opaque => {
+                                // We shouldn't get there
+                                register_error_or_panic!(
+                                    self.ctx,
+                                    st.meta.span.rust_span_data.span(),
+                                    "Unreachable case"
+                                );
+                                None
                             }
+                            TypeDeclKind::Error(_) => None,
+                            TypeDeclKind::Enum(variants) => Some(variants),
                         }
                     }
                 };
+                let Some(variants) = variants else {
+                    // An error occurred. We can't keep the `Rvalue::Discriminant` around so we
+                    // `Nop` the whole statement sequence.
+                    assert!(self.ctx.error_count > 0);
+                    *st = Statement {
```

Comment:
I think we should add an `Error` case in the statements. Could you put a TODO here and open an issue?